### PR TITLE
optimize: enable NamedThreadFactory to get ThreadGroup from the SecurityManager or Current thread

### PR DESCRIPTION
**Pull request by jxlgzwh**
Ⅰ. Describe what this PR did
enable NamedThreadFactory to get ThreadGroup from the SecurityManager or Current thread
Reference JDK's implementation of DefaultThreadFactory
Ⅱ. Does this pull request fix one issue?
Ⅲ. Why don't you add test cases (unit test/integration test)?
Ⅳ. Describe how to verify it
Ⅴ. Special notes for reviews

### CODE REVIEW

File: common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java [hunk 1]
```diff
@@ -65,4 +65,11 @@ public void testThreadNameHasCounterWithPrefixCounter() {
                 .isEqualTo(thread.getName());
         }
     }
+    @Test
```
Comment (selfishlover) [inline common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java:68]: Add an empty line before @Test would be better.

**Review by l81893521 — APPROVED** [review]
LGTM

File: common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java (hunk 1 above)
Comment (jxlgzwh) [reply common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java:69]: Thank you for your advice，I've dealt with it

Comment (jxlgzwh) [reply common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java:69]: > Add an empty line before @test would be better.
Thank you for your advice，I've dealt with it

**Review by jxlgzwh — COMMENTED** [review]
Add an empty line before @Test

File: common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java (hunk 1 above)
Comment (jxlgzwh) [reply common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java:69]: > Add an empty line before @test would be better.
Thank you for your suggestion. I have dealt with it

Comment (jxlgzwh) [reply common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java:69]: Thank you for your suggestion. I have dealt with it

Comment (jxlgzwh) [reply common/src/test/java/io/seata/common/thread/NamedThreadFactoryTest.java:69]: > Add an empty line before @test would be better.

Thank you for your suggestion. I have dealt with it

**Review by xingfudeshi — APPROVED** [review]
LGTM.

**Pull request merged**